Zed crashes on startup after installing language extension

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I can no longer start Zed. The last thing I did was install the swift language extension via the zed: extension registry.

# Crash report, partial:
```

Process:               zed [926]
Path:                  /Applications/Zed.app/Contents/MacOS/zed
Identifier:            dev.zed.Zed
Version:               0.123.4 (20240223.161253)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2024-02-23 13:32:31.3759 -0500
OS Version:            macOS 14.3.1 (23D60)
Report Version:        12
Anonymous UUID:        956FE670-AFBF-3A5F-7DC3-F0AF57908752


Time Awake Since Boot: 55 seconds

System Integrity Protection: enabled

Crashed Thread:        2  Dispatch queue: com.apple.root.user-initiated-qos

Exception Type:        EXC_BAD_ACCESS (SIGBUS)
Exception Codes:       KERN_PROTECTION_FAILURE at 0x0000000548400000
Exception Codes:       0x0000000000000002, 0x0000000548400000

Termination Reason:    Namespace SIGNAL, Code 10 Bus error: 10
Terminating Process:   exc handler [926]

VM Region Info: 0x548400000 is in 0x548400000-0x550000000;  bytes after start: 0  bytes before end: 130023423
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      VM_ALLOCATE                 548000000-548400000    [ 4096K] rw-/rwx SM=PRV  
--->  VM_ALLOCATE                 548400000-550000000    [124.0M] ---/rwx SM=NUL  
      VM_ALLOCATE                 550000000-558000000    [128.0M] ---/rwx SM=NUL  

Thread 0:: main Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	       0x1826c9874 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x1826dbcf0 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x1826d24b0 mach_msg_overwrite + 476
3   libsystem_kernel.dylib        	       0x1826c9bf8 mach_msg + 24
4   libdispatch.dylib             	       0x182573794 _dispatch_mach_send_and_wait_for_reply + 540
5   libdispatch.dylib             	       0x182573b24 dispatch_mach_send_with_result_and_wait_for_reply + 60
6   libxpc.dylib                  	       0x182429510 xpc_connection_send_message_with_reply_sync + 288
7   Foundation                    	       0x1838b3790 __NSXPCCONNECTION_IS_WAITING_FOR_A_SYNCHRONOUS_REPLY__ + 16
8   Foundation                    	       0x1838b1674 -[NSXPCConnection _sendInvocation:orArguments:count:methodSignature:selector:withProxy:] + 2168
9   CoreFoundation                	       0x1827c6530 ___forwarding___ + 1008
10  CoreFoundation                	       0x1827c6080 _CF_forwarding_prep_0 + 96
11  libFontRegistry.dylib         	       0x18acb6ce8 __51-[XTypeXPCClient XTCopyFontWithName:scope:options:]_block_invoke + 108
12  libFontRegistry.dylib         	       0x18acb3da4 -[XTypeXPCClient run:errorHandler:] + 128
13  libFontRegistry.dylib         	       0x18acb6c3c -[XTypeXPCClient XTCopyFontWithName:scope:options:] + 232
14  libFontRegistry.dylib         	       0x18aca26bc TGlobalFontRegistryImp::CopyFontForName(__CFString const*, TFontQueryOptions const&) const + 124
15  libFontRegistry.dylib         	       0x18aca672c TGlobalFontRegistry::CopyFontForName(__CFString const*, TFontQueryOptions const&) const + 172
16  libFontRegistry.dylib         	       0x18ac889cc XTCopyFontWithName + 144
17  CoreText                      	       0x184c11448 TDescriptorSource::CopyXTypeFontWithName(__CFString const*, unsigned long, unsigned long) + 124
18  CoreText                      	       0x184c0dcd4 TDescriptorSource::CopyFontDescriptorPerPostScriptName(__CFString const*, unsigned long, unsigned long, __CFString const*, __CFNumber const*, __CFNumber const*, CTFontLegibilityWeight, __CFBoolean const*) const + 524
19  CoreText                      	       0x184b50a78 TDescriptor::CreateMatchingDescriptorInternal(__CFSet const*, unsigned long) const + 1456
20  CoreText                      	       0x184b5f83c TDescriptor::InitBaseFont(unsigned long, double) + 76
21  CoreText                      	       0x184b50290 TDescriptor::CreateMatchingDescriptor(__CFSet const*, double, unsigned long) const + 448
22  CoreText                      	       0x184b500a8 CTFontDescriptorCreateMatchingFontDescriptor + 48
23  libMTLHud.dylib               	       0x222455c14 esfm_FindCTFont(char const*, unsigned int) + 108
24  libMTLHud.dylib               	       0x222456038 esfm_BakeFontIntoBitmapGrayscale + 120
25  libMTLHud.dylib               	       0x222456bc8 esfm_BakeFontIntoTextureR8Unorm + 244
26  libMTLHud.dylib               	       0x222456e9c esfm_CreateFont + 160
27  libMTLHud.dylib               	       0x2224575c4 Renderer::Init(CAMetalLayer*) + 160
28  libMTLHud.dylib               	       0x22245be74 Overlay::Init(CAMetalLayer*) + 356
29  libMTLHud.dylib               	       0x22245a048 invocation function for block in CAMetalLayer_nextDrawable(void (*)(), objc_object*, objc_selector*) + 52
30  libdispatch.dylib             	       0x182558910 _dispatch_client_callout + 20
31  libdispatch.dylib             	       0x18255a14c _dispatch_once_callout + 32
32  libMTLHud.dylib               	       0x222459d34 CAMetalLayer_nextDrawable(void (*)(), objc_object*, objc_selector*) + 264
33  libMTLHud.dylib               	       0x22245a14c invocation function for block in void replaceMethod<objc_object* __strong>(objc_class*, objc_selector*, objc_object* __strong (*)(void (*)(), objc_object*, objc_selector*)) + 60
34  zed                           	       0x101d69010 _RNvXs2_NtNtNtCs2mzBxgaCaKa_4gpui8platform3mac6windowNtB5_9MacWindowNtB9_14PlatformWindow4draw + 216
35  zed                           	       0x101d8dea4 _RNCNCNvMs9_NtCs2mzBxgaCaKa_4gpui6windowNtB9_6Window3new0s_0Bb_ + 332
36  zed                           	       0x101d8d844 _RNCNvMs9_NtCs2mzBxgaCaKa_4gpui6windowNtB7_6Window3new0B9_ + 2104
3
...
```


### Environment

I'm unable to start zed therefore I can not run that command palette action.

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.


```
2024-02-23T13:37:21-05:00 [INFO] ========== starting zed ==========
2024-02-23T13:37:21-05:00 [INFO] Opening main db
2024-02-23T13:37:21-05:00 [INFO] set environment variables from shell:/opt/homebrew/bin/fish, path:/opt/homebrew/opt/go@1.20/bin:/opt/homebrew/sbin:/Users/darrellroberts/go/bin:/Users/darrellroberts/.ghcup/bin:/Users/darrellroberts/Library/Android/sdk/platform-tools:/opt/homebrew/bin:/Users/darrellroberts/.local/bin:/Users/darrellroberts/.cargo/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin
2024-02-23T13:37:21-05:00 [ERROR] crates/settings/src/settings_file.rs:76: EOF while parsing a value at line 1 column 0
2024-02-23T13:37:21-05:00 [ERROR] crates/zed/src/zed.rs:524: EOF while parsing a value at line 1 column 0
2024-02-23T13:37:21-05:00 [INFO] Opening main db
2024-02-23T13:37:21-05:00 [INFO] build git repository ".git"
2024-02-23T13:37:21-05:00 [INFO] Opening main db
2024-02-23T13:37:21-05:00 [INFO] set status on client 0: Authenticating
2024-02-23T13:37:21-05:00 [INFO] Opening main db
2024-02-23T13:37:21-05:00 [INFO] set status on client 68593: Connecting
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -236,7 +236,7 @@ tiktoken-rs = "0.5.7"
 time = { version = "0.3", features = ["serde", "serde-well-known", "formatting"] }
 toml = "0.8"
 tower-http = "0.4.4"
-tree-sitter = { version = "0.21", features = ["wasm"] }
+tree-sitter = { version = ">= 0.20.100", features = ["wasm"] }
 tree-sitter-astro = { git = "https://github.com/virchau13/tree-sitter-astro.git", rev = "e924787e12e8a03194f36a113290ac11d6dc10f3" }
 tree-sitter-bash = { git = "https://github.com/tree-sitter/tree-sitter-bash", rev = "7331995b19b8f8aba2d5e26deb51d2195c18bc94" }
 tree-sitter-c = "0.20.1"
@@ -260,7 +260,7 @@ tree-sitter-haskell = { git = "https://github.com/tree-sitter/tree-sitter-haskel
 tree-sitter-hcl = { git = "https://github.com/MichaHoffmann/tree-sitter-hcl", rev = "v1.1.0" }
 tree-sitter-heex = { git = "https://github.com/phoenixframework/tree-sitter-heex", rev = "2e1348c3cf2c9323e87c2744796cf3f3868aa82a" }
 tree-sitter-html = "0.19.0"
-tree-sitter-json = { git = "https://github.com/tree-sitter/tree-sitter-json", rev = "40a81c01a40ac48744e0c8ccabbaba1920441199" }
+tree-sitter-json = { git = "https://github.com/tree-sitter/tree-sitter-json", rev = "3b129203f4b72d532f58e72c5310c0a7db3b8e6d" }
 tree-sitter-lua = "0.0.14"
 tree-sitter-markdown = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "330ecab87a3e3a7211ac69bbadc19eabecdb1cca" }
 tree-sitter-nix = { git = "https://github.com/nix-community/tree-sitter-nix", rev = "66e3e9ce9180ae08fc57372061006ef83f0abde7" }
@@ -290,7 +290,7 @@ which = "6.0.0"
 sys-locale = "0.3.1"
 
 [patch.crates-io]
-tree-sitter = { git = "https://github.com/tree-sitter/tree-sitter", rev = "9301c1e676e4c6a87ef6dba332cd2e1614a08333" }
+tree-sitter = { git = "https://github.com/tree-sitter/tree-sitter", rev = "da7063ba18381fd7dd3c7fc8e1bc43b129818569" }
 # Workaround for a broken nightly build of gpui: See #7644 and revisit once 0.5.3 is released.
 pathfinder_simd = { git = "https://github.com/servo/pathfinder.git", rev = "e4fcda0d5259d0acf902aee6de7d2501f2bd6629" }
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -236,7 +236,7 @@ tiktoken-rs = "0.5.7"
 time = { version = "0.3", features = ["serde", "serde-well-known", "formatting"] }
 toml = "0.8"
 tower-http = "0.4.4"
-tree-sitter = { version = "0.20", features = ["wasm"] }
+tree-sitter = { version = "0.21", features = ["wasm"] }
 tree-sitter-astro = { git = "https://github.com/virchau13/tree-sitter-astro.git", rev = "e924787e12e8a03194f36a113290ac11d6dc10f3" }
 tree-sitter-bash = { git = "https://github.com/tree-sitter/tree-sitter-bash", rev = "7331995b19b8f8aba2d5e26deb51d2195c18bc94" }
 tree-sitter-c = "0.20.1"
@@ -285,13 +285,12 @@ tree-sitter-zig = { git = "https://github.com/maxxnino/tree-sitter-zig", rev = "
 unindent = "0.1.7"
 url = "2.2"
 uuid = { version = "1.1.2", features = ["v4"] }
-wasmtime = "16"
+wasmtime = "18.0.1"
 which = "6.0.0"
 sys-locale = "0.3.1"
 
 [patch.crates-io]
-tree-sitter = { git = "https://github.com/tree-sitter/tree-sitter", rev = "1d8975319c2d5de1bf710e7e21db25b0eee4bc66" }
-wasmtime = { git = "https://github.com/bytecodealliance/wasmtime", rev = "v16.0.0" }
+tree-sitter = { git = "https://github.com/tree-sitter/tree-sitter", rev = "9301c1e676e4c6a87ef6dba332cd2e1614a08333" }
 # Workaround for a broken nightly build of gpui: See #7644 and revisit once 0.5.3 is released.
 pathfinder_simd = { git = "https://github.com/servo/pathfinder.git", rev = "e4fcda0d5259d0acf902aee6de7d2501f2bd6629" }
 

--- a/crates/extension/src/extension_store.rs
+++ b/crates/extension/src/extension_store.rs
@@ -406,10 +406,6 @@ impl ExtensionStore {
             }));
 
         for language_name in &languages_to_add {
-            if language_name.as_ref() == "Swift" {
-                continue;
-            }
-
             let language = manifest.languages.get(language_name.as_ref()).unwrap();
             let mut language_path = self.extensions_dir.clone();
             language_path.extend([language.extension.as_ref(), language.path.as_path()]);

